src/main/scala/io/iohk/ethereum/keystore/KeyStore.scala

```diff
@@ -0,0 +1,90 @@
+package io.iohk.ethereum.keystore
+
+import java.io.File
+import java.nio.file.{Files, Paths}
+
+import akka.util.ByteString
+import io.iohk.ethereum.crypto._
+import io.iohk.ethereum.domain.Address
+import org.spongycastle.util.encoders.Hex
+
+import scala.util.Try
+
+
+object KeyStore {
+  sealed trait KeyStoreError
+  case object WrongPassphrase extends KeyStoreError
+  case object KeyNotFound extends KeyStoreError
+  case class IOError(msg: String) extends KeyStoreError
+}
+
+import io.iohk.ethereum.keystore.KeyStore._
+
+trait KeyStore {
+  def newAccount(passphrase: String): Either[KeyStoreError, Address]
+
+  def importPrivateKey(key: Array[Byte], passphrase: String): Either[KeyStoreError, Address]
+
+  def listAccounts(): Either[KeyStoreError, List[Address]]
+
+  def unlockAccount(address: Address, passphrase: String): Either[KeyStoreError, Wallet]
+}
+
+class KeyStoreImpl(keyStoreDir: String) extends KeyStore {
+
+  def newAccount(passphrase: String): Either[KeyStoreError, Address] = {
+    val keyPair = generateKeyPair()
+    val (prvKey, pubKey) = keyPairToByteArrays(keyPair)
+    val address = Address(kec256(pubKey))
+    save(address, prvKey, passphrase).map(_ => address)
+  }
+
+  def importPrivateKey(prvKey: Array[Byte], passphrase: String): Either[KeyStoreError, Address] = {
+    val pubKey = pubKeyFromPrvKey(prvKey)
+    val address = Address(kec256(pubKey))
+    save(address, prvKey, passphrase).map(_ => address)
+  }
+
+  def listAccounts(): Either[KeyStoreError, List[Address]] = {
+    val dir = new File(keyStoreDir)
+    Try {
+      if (!dir.exists() || !dir.isDirectory())
```

Comment:
I'm wondering, should we return an empty list if `dir` wasn't created yet? It seems currently return `"Could not read /tmp/etc-client/keystore"` if no account was created, and geth  returns an empty list in that scenario.